source/parsing/Lexer.cpp

```diff
@@ -1198,6 +1198,96 @@ void Lexer::scanWhitespace() {
     addTrivia(TriviaKind::Whitespace);
 }
 
+bool detectTranslateOnOffPragma(std::string_view view, bool offMode) {
+    static std::vector<std::string_view> offCandidates = {
```

Comment:
These should be moved out of the function and marked const; statics inside of functions require the compiler to insert code to make sure statics get atomically initialized the first time the function is called, which is an unnecessary expense.

Reply:
Also you should clearly document that the ordering here is really important, since your algorithm relies on them being in alphabetical order.

Reply:
I've moved the trigger word list, and documented the hidden requirements